v03_pipeline/lib/misc/allele_registry.py

```diff
@@ -0,0 +1,145 @@
+import dataclasses
+import hashlib
+import math
+import tempfile
+import time
+
+import hail as hl
+import requests
+from requests import HTTPError
+
+from v03_pipeline.lib.logger import get_logger
+from v03_pipeline.lib.model import Env, ReferenceGenome
+
+MAX_VARIANTS_PER_REQUEST = 1000000
+ALLELE_REGISTRY_URL = 'https://reg.genome.network/alleles?file=vcf&fields=none+@id+externalRecords.gnomAD_4.id'
+HTTP_REQUEST_TIMEOUT = 300
+
+logger = get_logger(__name__)
+
+
+@dataclasses.dataclass
+class AlleleRegistryError:
+    base_url: str
+    error_type: str
+    description: str
+    message: str
+    input_line: str | None
+
+    @classmethod
+    def from_api_response(cls, response: dict, base_url: str):
+        return cls(
+            base_url=base_url,
+            error_type=response['errorType'],
+            description=response['description'],
+            message=response['message'],
+            input_line=response.get('inputLine'),
+        )
+
+    @property
+    def loggable_message(self) -> str:
```

Comment:
Would just implementing `__str__` or `__rpr__` work for this?

Reply:
`__str__` is correct here because the logger will convert an object to a string, so I can just do `logger.error(e)` where `e` is an instance of `AlleleRegistryError`. thanks for the suggestion!